src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +18,36 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (HibernateException e) {
+            throw new DataProcessingException("Cannot open the session", e);
+        } catch (EntityExistsException | TransactionRequiredException e) {
+            throw new DataProcessingException("Cannot save the actor: " + actor, e);
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Commit fails.", e);
```

Comment:
The exception message 'Commit fails.' could be more descriptive. Consider including the actor's details in the message to provide more context about the failure.